<!DOCTYPE html><html lang="en"><head><meta charSet="utf-8"/><meta name="viewport" content="width=device-width, initial-scale=1"/><link rel="icon" href="/images/svgs/logo4_.svg" type="image/svg+xml"/><title>TeraLab | Research Area</title><meta name="description" content="Research Area"/><meta name="next-head-count" content="5"/><link rel="preload" href="/_next/static/media/4f05ba3a6752a328-s.p.woff2" as="font" type="font/woff2" crossorigin="anonymous" data-next-font="size-adjust"/><script id="theme-switcher" data-nscript="beforeInteractive">
 if (
  localStorage.getItem('theme') === 'dark' ||
  (!('theme' in localStorage) &&
    window.matchMedia('(prefers-color-scheme: dark)').matches)
) {
  document.documentElement.classList.add('dark');
} else {
  document.documentElement.classList.remove('dark');
}
  </script><link rel="preload" href="/_next/static/css/03a3ed200f118521.css" as="style"/><link rel="stylesheet" href="/_next/static/css/03a3ed200f118521.css" data-n-g=""/><noscript data-n-css=""></noscript><script defer="" nomodule="" src="/_next/static/chunks/polyfills-78c92fac7aa8fdd8.js"></script><script src="/_next/static/chunks/webpack-8fa1640cc84ba8fe.js" defer=""></script><script src="/_next/static/chunks/framework-305cb810cde7afac.js" defer=""></script><script src="/_next/static/chunks/main-67df5d518c5e17f9.js" defer=""></script><script src="/_next/static/chunks/pages/_app-0d43363ab381585e.js" defer=""></script><script src="/_next/static/chunks/pages/research-7a0b5c7085acceb3.js" defer=""></script><script src="/_next/static/o9QPg0C9odxtMLOKKBLVm/_buildManifest.js" defer=""></script><script src="/_next/static/o9QPg0C9odxtMLOKKBLVm/_ssgManifest.js" defer=""></script></head><body><div id="__next"><main class="__variable_4bc053 font-mont bg-light dark:bg-dark w-full min-h-screen h-full"><header class="fixed w-full items-center font-medium z-50
  xl:grid xl:grid-cols-3
  flex justify-between px-6 py-4 md:px-12 xl:px-32
  transition-colors duration-300
  bg-transparent text-dark text-dark dark:text-light"><button type="button" class=" flex-col items-center justify-center hidden xl:flex xl:col-start-2 " aria-controls="mobile-menu" aria-expanded="false"><span class="sr-only">Open main menu</span><span class="bg-dark dark:bg-light block h-0.5 w-6 rounded-sm transition-all duration-300 ease-out -translate-y-0.5"></span><span class="bg-dark dark:bg-light block h-0.5 w-6 rounded-sm transition-all duration-300 ease-out opacity-100 my-0.5"></span><span class="bg-dark dark:bg-light block h-0.5 w-6 rounded-sm transition-all duration-300 ease-out translate-y-0.5"></span></button><div class="flex items-end xl:hidden"><div class=""><a href="/"><img alt="TeraLab logo" viewBox="0 0 24 24" loading="lazy" width="1000" height="705" decoding="async" data-nimg="1" class="h-16 w-16 " style="color:transparent" src="/_next/static/media/logo4_.e09df80e.svg"/></a></div><div class="">TeraLab</div></div><nav class="flex items-center justify-center xl:hidden"><a class="mx-4  rounded relative group xl:text-light xl:dark:text-dark" href="/">Home<span class="
              inline-block h-[1px] bg-dark absolute left-0 -bottom-0.5 
              group-hover:w-full transition-[width] ease duration-300 dark:bg-light
               w-0 xl:bg-light xl:dark:bg-dark
              "> </span></a><a class="mx-4  rounded relative group xl:text-light xl:dark:text-dark" href="/people">People<span class="
              inline-block h-[1px] bg-dark absolute left-0 -bottom-0.5 
              group-hover:w-full transition-[width] ease duration-300 dark:bg-light
               w-0 xl:bg-light xl:dark:bg-dark
              "> </span></a><a class="mx-4 whitespace-nowrap  rounded relative group xl:text-light xl:dark:text-dark" href="/research">Research Area<span class="
              inline-block h-[1px] bg-dark absolute left-0 -bottom-0.5 
              group-hover:w-full transition-[width] ease duration-300 dark:bg-light
              w-full xl:bg-light xl:dark:bg-dark
              "> </span></a><a class="mx-4  rounded relative group xl:text-light xl:dark:text-dark" href="/publications">Publications<span class="
              inline-block h-[1px] bg-dark absolute left-0 -bottom-0.5 
              group-hover:w-full transition-[width] ease duration-300 dark:bg-light
               w-0 xl:bg-light xl:dark:bg-dark
              "> </span></a><a class="mx-4  rounded relative group xl:text-light xl:dark:text-dark" href="/projects">Projects<span class="
              inline-block h-[1px] bg-dark absolute left-0 -bottom-0.5 
              group-hover:w-full transition-[width] ease duration-300 dark:bg-light
               w-0 xl:bg-light xl:dark:bg-dark
              "> </span></a><a class="mx-4  rounded relative group xl:text-light xl:dark:text-dark" href="/events">Events<span class="
              inline-block h-[1px] bg-dark absolute left-0 -bottom-0.5 
              group-hover:w-full transition-[width] ease duration-300 dark:bg-light
               w-0 xl:bg-light xl:dark:bg-dark
              "> </span></a><a class="mx-4  rounded relative group xl:text-light xl:dark:text-dark" href="/news">News<span class="
              inline-block h-[1px] bg-dark absolute left-0 -bottom-0.5 
              group-hover:w-full transition-[width] ease duration-300 dark:bg-light
               w-0 xl:bg-light xl:dark:bg-dark
              "> </span></a></nav><nav class="flex items-center justify-end flex-wrap xl:mt-2 xl:hidden "><a target="_blank" class="w-6 mr-3" href="#" aria-label="Checkout my twitter profile" tabindex="0"><svg xmlns="http://www.w3.org/2000/svg" width="1em" height="1em" viewBox="0 0 1200 1400" class="w-full h-auto undefined"><path fill="currentColor" d="M714.163 519.284 1160.89 0H1052.64L670.95 442.001 369.75 0H0l468.908 681.821L0 1226.27h108.25l401.1-476.54 316.401 476.54H1200L714.137 519.284h.026ZM557.602 687.828l-46.498-66.776-369.75-530.04h158.97l298.542 427.831 46.498 66.775 392.726 563.344h-158.97L557.602 687.828Z"></path></svg></a><a target="_blank" class="w-6 mx-3" href="#" aria-label="Checkout my linkedin profile" tabindex="0"><svg xmlns="http://www.w3.org/2000/svg" width="1em" height="1em" viewBox="0 0 256 256" class="w-full h-auto undefined"><path fill="none" d="M0 0h256v256H0z"></path><g fill="none"><rect width="256" height="256" fill="#fff" rx="60"></rect><rect width="256" height="256" fill="#0A66C2" rx="60"></rect><path fill="#fff" d="M184.715 217.685h29.27a4 4 0 0 0 4-3.999l.015-61.842c0-32.323-6.965-57.168-44.738-57.168-14.359-.534-27.9 6.868-35.207 19.228a.32.32 0 0 1-.595-.161V101.66a4 4 0 0 0-4-4h-27.777a4 4 0 0 0-4 4v112.02a4 4 0 0 0 4 4h29.268a4 4 0 0 0 4-4v-55.373c0-15.657 2.97-30.82 22.381-30.82 19.135 0 19.383 17.916 19.383 31.834v54.364a4 4 0 0 0 4 4ZM38 59.627c0 11.865 9.767 21.627 21.632 21.627 11.862-.001 21.623-9.769 21.623-21.631C81.253 47.761 71.491 38 59.628 38 47.762 38 38 47.763 38 59.627Zm6.959 158.058h29.307a4 4 0 0 0 4-4V101.66a4 4 0 0 0-4-4H44.959a4 4 0 0 0-4 4v112.025a4 4 0 0 0 4 4Z"></path></g></svg></a><button class="w-6 h-6 ease ml-3 flex items-center justify-center rounded-full p-1  
            bg-dark  text-light
            " aria-label="theme-switcher"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" class="w-full h-auto fill-dark"><rect x="0" y="0" width="24" height="24" fill="rgba(255, 255, 255, 0)"></rect><g fill="none" stroke="currentColor" stroke-dasharray="2" stroke-dashoffset="2" stroke-linecap="round" stroke-width="2"><path d="M0 0"><animate fill="freeze" attributeName="d" begin="1.2s" dur="0.2s" values="M12 19v1M19 12h1M12 5v-1M5 12h-1;M12 21v1M21 12h1M12 3v-1M3 12h-1"></animate><animate fill="freeze" attributeName="stroke-dashoffset" begin="1.2s" dur="0.2s" values="2;0"></animate></path><path d="M0 0"><animate fill="freeze" attributeName="d" begin="1.5s" dur="0.2s" values="M17 17l0.5 0.5M17 7l0.5 -0.5M7 7l-0.5 -0.5M7 17l-0.5 0.5;M18.5 18.5l0.5 0.5M18.5 5.5l0.5 -0.5M5.5 5.5l-0.5 -0.5M5.5 18.5l-0.5 0.5"></animate><animate fill="freeze" attributeName="stroke-dashoffset" begin="1.5s" dur="1.2s" values="2;0"></animate></path><animateTransform attributeName="transform" dur="30s" repeatCount="indefinite" type="rotate" values="0 12 12;360 12 12"></animateTransform></g><g fill="currentColor"><path d="M15.22 6.03L17.75 4.09L14.56 4L13.5 1L12.44 4L9.25 4.09L11.78 6.03L10.87 9.09L13.5 7.28L16.13 9.09L15.22 6.03Z"><animate fill="freeze" attributeName="fill-opacity" dur="0.4s" values="1;0"></animate></path><path d="M19.61 12.25L21.25 11L19.19 10.95L18.5 9L17.81 10.95L15.75 11L17.39 12.25L16.8 14.23L18.5 13.06L20.2 14.23L19.61 12.25Z"><animate fill="freeze" attributeName="fill-opacity" begin="0.2s" dur="0.4s" values="1;0"></animate></path></g><g fill="currentColor" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="2"><path d="M7 6 C7 12.08 11.92 17 18 17 C18.53 17 19.05 16.96 19.56 16.89 C17.95 19.36 15.17 21 12 21 C7.03 21 3 16.97 3 12 C3 8.83 4.64 6.05 7.11 4.44 C7.04 4.95 7 5.47 7 6 Z"></path><set attributeName="opacity" begin="0.6s" to="0"></set></g><mask id="lineMdMoonFilledToSunnyFilledLoopTransition0"><circle cx="12" cy="12" r="12" fill="#fff"></circle><circle cx="18" cy="6" r="12" fill="#fff"><animate fill="freeze" attributeName="cx" begin="0.6s" dur="0.4s" values="18;22"></animate><animate fill="freeze" attributeName="cy" begin="0.6s" dur="0.4s" values="6;2"></animate><animate fill="freeze" attributeName="r" begin="0.6s" dur="0.4s" values="12;3"></animate></circle><circle cx="18" cy="6" r="10"><animate fill="freeze" attributeName="cx" begin="0.6s" dur="0.4s" values="18;22"></animate><animate fill="freeze" attributeName="cy" begin="0.6s" dur="0.4s" values="6;2"></animate><animate fill="freeze" attributeName="r" begin="0.6s" dur="0.4s" values="10;1"></animate></circle></mask><circle cx="12" cy="12" r="10" fill="currentColor" mask="url(#lineMdMoonFilledToSunnyFilledLoopTransition0)" opacity="0"><set attributeName="opacity" begin="0.6s" to="1"></set><animate fill="freeze" attributeName="r" begin="0.6s" dur="0.4s" values="10;6"></animate></circle></svg></button></nav></header><div class="fixed top-0 left-0 w-full h-full bg-light dark:bg-dark z-30 pointer-events-none" style="opacity:0;transform:none"></div><main class="mb-16 flex w-full flex-col items-center justify-center dark:text-light"><div class="mt-16 z-0 inline-block h-full w-full bg-light p-32 dark:bg-dark xl:p-24 lg:p-16 
      md:p-12 sm:p-8 pt-16"><h2 class="font-bold text-4xl w-full text-center my-16">Research Area</h2><div class="grid grid-cols-12 gap-24 gap-y-2 xl:gap-x-16 lg:gap-x-8 md:gap-y-24 sm:gap-x-0"><div class="col-span-12"><li class="flex w-full max-w-4xl mx-auto items-center justify-center gap-6 py-6 border-b border-gray-300 dark:border-gray-700" style="transform:none"><div class="flex-shrink-0 w-52 h-12 rounded-full overflow-hidden md:hidden"><a target="_blank" rel="noopener noreferrer" href="https://riccardoc95.github.io/PixHomology/"><img src="images/research/pixhomology.png" alt="Topological Data Analysis of Images via Persistent Homology" class="w-full h-full object-cover"/></a></div><div class="flex flex-col justify-center text-left w-full"><a target="_blank" class="hover:underline underline-offset-4" href="https://riccardoc95.github.io/PixHomology/"><h2 class="text-xl font-bold text-dark dark:text-light">Topological Data Analysis of Images via Persistent Homology</h2></a><a target="_blank" rel="noopener noreferrer" href="https://riccardoc95.github.io/PixHomology/"><img src="images/research/pixhomology.png" alt="Topological Data Analysis of Images via Persistent Homology" class="w-full h-full object-cover hidden md:flex"/></a><span class="text-sm text-primary dark:text-primaryDark mb-1">Persistent Homology, Topological Data Analysis, Image Topology</span><p class="text-sm text-gray-700 dark:text-gray-300 max-w-xl">This research area focuses on the development and application of topological data analysis (TDA) techniques, specifically persistent homology, for the study of 2D images. By leveraging topological invariants, the goal is to extract robust and scale-invariant features that capture the underlying structure of image data. The approach enables a deeper understanding of shape, connectivity, and texture beyond traditional pixel-based methods.</p><div class="mt-3 flex items-center gap-4"><a target="_blank" aria-label="Topological Data Analysis of Images via Persistent Homology GitHub link" href="https://github.com/riccardoc95/PixHomology"><svg xmlns="http://www.w3.org/2000/svg" width="1em" height="1em" viewBox="0 0 512 512" class="w-full h-auto w-5 h-5 hover:text-gray-800 dark:hover:text-gray-200"><path fill="none" d="M0 0h512v512H0z"></path><path fill="currentColor" d="M256 32C132.3 32 32 134.9 32 261.7c0 101.5 64.2 187.5 153.2 217.9a17.56 17.56 0 0 0 3.8.4c8.3 0 11.5-6.1 11.5-11.4 0-5.5-.2-19.9-.3-39.1a102.4 102.4 0 0 1-22.6 2.7c-43.1 0-52.9-33.5-52.9-33.5-10.2-26.5-24.9-33.6-24.9-33.6-19.5-13.7-.1-14.1 1.4-14.1h.1c22.5 2 34.3 23.8 34.3 23.8 11.2 19.6 26.2 25.1 39.6 25.1a63 63 0 0 0 25.6-6c2-14.8 7.8-24.9 14.2-30.7-49.7-5.8-102-25.5-102-113.5 0-25.1 8.7-45.6 23-61.6-2.3-5.8-10-29.2 2.2-60.8a18.64 18.64 0 0 1 5-.5c8.1 0 26.4 3.1 56.6 24.1a208.21 208.21 0 0 1 112.2 0c30.2-21 48.5-24.1 56.6-24.1a18.64 18.64 0 0 1 5 .5c12.2 31.6 4.5 55 2.2 60.8 14.3 16.1 23 36.6 23 61.6 0 88.2-52.4 107.6-102.3 113.3 8 7.1 15.2 21.1 15.2 42.5 0 30.7-.3 55.5-.3 63 0 5.4 3.1 11.5 11.4 11.5a19.35 19.35 0 0 0 4-.4C415.9 449.2 480 363.1 480 261.7 480 134.9 379.7 32 256 32Z"></path></svg></a><a target="_blank" class="text-sm font-semibold text-primary dark:text-primaryDark hover:underline" aria-label="Visit Topological Data Analysis of Images via Persistent Homology" href="https://riccardoc95.github.io/PixHomology/">Read more →</a></div></div></li></div><div class="col-span-12"><li class="flex w-full max-w-4xl mx-auto items-center justify-center gap-6 py-6 border-b border-gray-300 dark:border-gray-700" style="transform:none"><div class="flex-shrink-0 w-52 h-12 rounded-full overflow-hidden md:hidden"><a target="_blank" rel="noopener noreferrer" href="/research"><img src="images/research/blank.png" alt="High-Performance Computational Genomics" class="w-full h-full object-cover"/></a></div><div class="flex flex-col justify-center text-left w-full"><a target="_blank" class="hover:underline underline-offset-4" href="/research"><h2 class="text-xl font-bold text-dark dark:text-light">High-Performance Computational Genomics</h2></a><a target="_blank" rel="noopener noreferrer" href="/research"><img src="images/research/blank.png" alt="High-Performance Computational Genomics" class="w-full h-full object-cover hidden md:flex"/></a><span class="text-sm text-primary dark:text-primaryDark mb-1">Computational Genomics, High-Performance Computing, Bioinformatics</span><p class="text-sm text-gray-700 dark:text-gray-300 max-w-xl">This research area explores the intersection of large-scale genomics and high-performance computing (HPC), focusing on the design of scalable and efficient computational workflows. Core topics include distributed computing application for variant detection across both linear and pangenomic reference models, and the development of accelerated pipelines for long-read sequencing data, including error correction. By harnessing parallel architectures and optimized algorithms, the goal is to extend the limits of genomic data processing, enabling faster, deeper, and more comprehensive analyses at unprecedented scales.</p><div class="mt-3 flex items-center gap-4"></div></div></li></div><div class="col-span-12"><li class="flex w-full max-w-4xl mx-auto items-center justify-center gap-6 py-6 border-b border-gray-300 dark:border-gray-700" style="transform:none"><div class="flex-shrink-0 w-52 h-12 rounded-full overflow-hidden md:hidden"></div><div class="flex flex-col justify-center text-left w-full"><a target="_blank" class="hover:underline underline-offset-4" href="/research"><h2 class="text-xl font-bold text-dark dark:text-light">Advanced Computational Methods for Genomics</h2></a><span class="text-sm text-primary dark:text-primaryDark mb-1">Computational Biology, , Genomic Data Representation, Deep Learning</span><p class="text-sm text-gray-700 dark:text-gray-300 max-w-xl">This research area focuses on the analysis of different genomic data representations (from sequences to graphs) using approaches that extend beyond traditional combinatorial algorithms. The goal is to improve the depth, accuracy, and scalability of genomic analyses, ranging from querying large-scale genomic databases embedded in learned latent spaces to developing novel assembly techniques that exploit the geometry of assembly graphs. This work combines innovative mathematical formulations of genomic problems with state-of-the-art deep learning models to advance both theory and practical applications in genomics.</p><div class="mt-3 flex items-center gap-4"></div></div></li></div><div class="col-span-12"><li class="flex w-full max-w-4xl mx-auto items-center justify-center gap-6 py-6 border-b border-gray-300 dark:border-gray-700" style="transform:none"><div class="flex-shrink-0 w-52 h-12 rounded-full overflow-hidden md:hidden"></div><div class="flex flex-col justify-center text-left w-full"><a target="_blank" class="hover:underline underline-offset-4" href="/research"><h2 class="text-xl font-bold text-dark dark:text-light">Domain-Aware LLMs with RAG and Knowledge Graphs</h2></a><span class="text-sm text-primary dark:text-primaryDark mb-1"></span><p class="text-sm text-gray-700 dark:text-gray-300 max-w-xl">This research area explores the intersection of large language models (LLMs), knowledge engineering, and retrieval-augmented architectures, focusing on the specialization of LLMs for domain-specific tasks. Core topics include the construction of domain knowledge graphs and their integration into Knowledge-Augmented Generation (KAG) frameworks, the design of hybrid Retrieval-Augmented Generation (RAG) pipelines combining vector and graph-based retrieval, and the fine-tuning of LLMs on curated corpora. These techniques aim to enhance the domain awareness, factual grounding, and adaptability of generative systems across diverse application contexts.</p><div class="mt-3 flex items-center gap-4"></div></div></li></div></div></div></main><footer class="w-full border-t-2 border-solid border-dark dark:border-light py-6 px-4 text-lg font-medium dark:text-light"><div class="max-w-7xl mx-auto flex items-center justify-between flex-wrap gap-4 sm:flex-col sm:text-center"><div class="flex items-center gap-2">TeraLab @<!-- --> <a href="https://www.dss.uniroma1.it/">Dipartimento di Scienze Statistiche - Sapienza</a></div><div class="text-sm text-gray-600 dark:text-gray-400 sm:mt-2"><span class="text-sm sm:text-base"> © <!-- -->2025<!-- --> All Rights Reserved</span></div></div></footer></main></div><script id="__NEXT_DATA__" type="application/json">{"props":{"pageProps":{"data":[{"keywords":"Persistent Homology, Topological Data Analysis, Image Topology","title":"Topological Data Analysis of Images via Persistent Homology","summary":"This research area focuses on the development and application of topological data analysis (TDA) techniques, specifically persistent homology, for the study of 2D images. By leveraging topological invariants, the goal is to extract robust and scale-invariant features that capture the underlying structure of image data. The approach enables a deeper understanding of shape, connectivity, and texture beyond traditional pixel-based methods.","img":"images/research/pixhomology.png","link":"https://riccardoc95.github.io/PixHomology/","github":"https://github.com/riccardoc95/PixHomology"},{"keywords":"Computational Genomics, High-Performance Computing, Bioinformatics","title":"High-Performance Computational Genomics","summary":"This research area explores the intersection of large-scale genomics and high-performance computing (HPC), focusing on the design of scalable and efficient computational workflows. Core topics include distributed computing application for variant detection across both linear and pangenomic reference models, and the development of accelerated pipelines for long-read sequencing data, including error correction. By harnessing parallel architectures and optimized algorithms, the goal is to extend the limits of genomic data processing, enabling faster, deeper, and more comprehensive analyses at unprecedented scales.","img":"images/research/blank.png","link":"","github":""},{"keywords":"Computational Biology, , Genomic Data Representation, Deep Learning","title":"Advanced Computational Methods for Genomics","summary":"This research area focuses on the analysis of different genomic data representations (from sequences to graphs) using approaches that extend beyond traditional combinatorial algorithms. The goal is to improve the depth, accuracy, and scalability of genomic analyses, ranging from querying large-scale genomic databases embedded in learned latent spaces to developing novel assembly techniques that exploit the geometry of assembly graphs. This work combines innovative mathematical formulations of genomic problems with state-of-the-art deep learning models to advance both theory and practical applications in genomics.","img":"","link":"","github":""},{"keywords":"","title":"Domain-Aware LLMs with RAG and Knowledge Graphs","summary":"This research area explores the intersection of large language models (LLMs), knowledge engineering, and retrieval-augmented architectures, focusing on the specialization of LLMs for domain-specific tasks. Core topics include the construction of domain knowledge graphs and their integration into Knowledge-Augmented Generation (KAG) frameworks, the design of hybrid Retrieval-Augmented Generation (RAG) pipelines combining vector and graph-based retrieval, and the fine-tuning of LLMs on curated corpora. These techniques aim to enhance the domain awareness, factual grounding, and adaptability of generative systems across diverse application contexts.","img":"","link":"","github":""}]},"__N_SSG":true},"page":"/research","query":{},"buildId":"o9QPg0C9odxtMLOKKBLVm","isFallback":false,"gsp":true,"scriptLoader":[]}</script></body></html>